components/scream/src/interface/atm_comp_mct.F90

```diff
@@ -36,36 +63,104 @@ subroutine scream_init(f_comm,start_ymd,start_tod) bind(c)
-    character(len=*), optional,   intent(IN)    :: NLFilename ! Namelist filename
-    !
-    ! Locals
-    !
-    integer :: f_mpicomm
-    integer (kind=SHR_KIND_IN) :: start_tod, start_ymd
+    ! !DESCRIPTION: initialize dead atm model
+
+    ! !INPUT/OUTPUT PARAMETERS:
+    type(ESMF_Clock)            , intent(inout) :: EClock
+    type(seq_cdata)             , intent(inout) :: cdata
+    type(mct_aVect)             , intent(inout) :: x2d, d2x
+    character(len=*), optional  , intent(in)    :: NLFilename ! Namelist filename
+
+    !--- local variables ---
+    type(seq_infodata_type), pointer :: infodata
+    type(mct_gsMap)        , pointer :: gsMap
+    type(mct_gGrid)        , pointer :: ggrid
+    integer(IN)                      :: shrlogunit     ! original log unit
+    integer(IN)                      :: shrloglev      ! original log level
+    integer(IN)                      :: nxg            ! global dim i-direction
+    integer(IN)                      :: nyg            ! global dim j-direction
+    integer(IN)                      :: phase          ! initialization phase
+    integer(IN)                      :: ierr           ! error code
+    logical                          :: atm_present    ! if true, component is present
+    logical                          :: atm_prognostic ! if true, component is prognostic
+    integer (kind=SHR_KIND_IN)       :: start_tod, start_ymd
+    !-------------------------------------------------------------------------------
+
+    ! Set cdata pointers to derived types (in coupler)
+    call seq_cdata_setptrs(cdata, &
+         id=compid, &
+         mpicom=mpicom, &
+         gsMap=gsmap, &
+         dom=ggrid, &
+         infodata=infodata)
+
+    ! Obtain infodata variables
+    call seq_infodata_getData(infodata, atm_phase=phase)
+    if (phase > 1) RETURN
+
+    ! Determine instance information
+    inst_name   = seq_comm_name(compid)
+    inst_index  = seq_comm_inst(compid)
+    inst_suffix = seq_comm_suffix(compid)
+
+    if (phase == 1) then
+       ! Determine communicator group
+       call mpi_comm_rank(mpicom, my_task, ierr)
+
+       !--- open log file ---
+       if (my_task == master_task) then
+          logUnit = shr_file_getUnit()
+          call shr_file_setIO('atm_modelio.nml'//trim(inst_suffix),logUnit)
+       else
+          logUnit = 6
+       endif
+    endif
+
+    !----------------------------------------------------------------------------
+    ! Reset shr logging to my log file
+    !----------------------------------------------------------------------------
+
+    call shr_file_getLogUnit (shrlogunit)
+    call shr_file_getLogLevel(shrloglev)
+    call shr_file_setLogUnit (logUnit)
+
+    !----------------------------------------------------------------------------
+    ! Initialize xatm
+    !----------------------------------------------------------------------------
+
+    call dead_init_mct('atm', Eclock, x2d, d2x, &
```

Comment:
Are the dead_xyz calls there only for now, to get things working, or do they perform some common operations (common meaning needed in both a dead_atm setup and an active_atm setup)?

If they will always be used, we should rename the dead_xyz to remove 'dead'. If they are used only in dead_atm runs, we should guard them with some compile-time ifdef stuff. Or, at the very least, we should put some comment that remind the user that this stuff is there only for dead_atm runs, and it should be skipped/avoided in active_atm runs.

Reply:
Yes, they are essentially place holders for now.  Once we are ready to turn ATM on I think we will be changing them.  Since I'm not yet an expert on the MCT coupling I'm inclined to leave them as is.